packages/selfcare-client-users-updater/test/selfcareClientUsersUpdaterProcessor.test.ts

```diff
@@ -0,0 +1,262 @@
+import { relationshipStatus } from "../src/model/UsersEventPayload.js";
+import { readModelServiceBuilder } from "../src/services/readModelService.js";
+import {
+  correctEventPayload,
+  generateInternalTokenMock,
+  kafkaMessagePayload,
+} from "./utils.js";
+
+export const { cleanup, readModelRepository, postgresDB } =
+  await setupTestContainersVitest(
+    inject("readModelConfig"),
+    inject("eventStoreConfig")
+  );
+
+afterEach(cleanup);
+
+describe("selfcareClientUsersUpdaterProcessor", () => {
+  const readModelService = readModelServiceBuilder(readModelRepository);
+
+  async function seedCollection(data: Array<{ data: Client }>): Promise<void> {
+    await readModelRepository.clients.insertMany(data as never);
+  }
+
+  const userId: UUID = randomUUID();
+  const clientMock = {
+    ...getMockClient(),
+    adminId: userId,
+    kind: clientKind.api,
+  };
+  const clientMock2 = {
+    ...getMockClient(),
+    adminId: userId,
+    kind: clientKind.api,
+  };
+  const authorizationProcessClientMock = {
+    client: {
+      internalRemoveClientAdmin: vi.fn().mockResolvedValue(undefined),
+    },
+  } as unknown as AuthorizationProcessClient;
+  const tokenGeneratorMock = new InteropTokenGenerator(config);
+  const refreshableTokenMock = new RefreshableInteropToken(tokenGeneratorMock);
+  let selfcareClientUsersUpdaterProcessor: ReturnType<
+    typeof selfcareClientUsersUpdaterProcessorBuilder
+  >;
+
+  beforeAll(async () => {
+    selfcareClientUsersUpdaterProcessor =
+      selfcareClientUsersUpdaterProcessorBuilder(
+        refreshableTokenMock,
+        authorizationProcessClientMock,
+        readModelService,
+        config.interopProduct
+      );
+  });
+
+  let refreshableInternalTokenSpy: MockInstance;
+  beforeEach(() => {
+    vi.spyOn(tokenGeneratorMock, "generateInternalToken").mockImplementation(
+      generateInternalTokenMock
+    );
+    refreshableInternalTokenSpy = vi
+      .spyOn(refreshableTokenMock, "get")
+      .mockImplementation(generateInternalTokenMock);
+  });
+
+  afterEach(async () => {
+    vi.clearAllMocks();
+  });
+
+  it.each([relationshipStatus.suspended, relationshipStatus.deleted])(
+    "should remove admin when user has relationshipStatus %s",
+    async (status) => {
+      const message: EachMessagePayload = {
+        ...kafkaMessagePayload,
+        message: {
+          ...kafkaMessagePayload.message,
+          value: Buffer.from(
+            JSON.stringify({
+              ...correctEventPayload,
+              productId: config.interopProduct,
+              user: {
+                userId,
+                name: "Test Name",
+                familyName: "Test FamilyName",
+                email: "test@example.com",
+                role: "Test Role",
+                productRole: userRole.ADMIN_ROLE,
+                relationshipStatus: status,
+                mobilePhone: "1234567890",
```

Comment:
```suggestion
              user: {
                ...correctEventPayload.user,
                productRole: userRole.ADMIN_ROLE,
                relationshipStatus: status,
              },
```
let's just add the fields we are interested in